handler/panos.go

```diff
@@ -138,18 +144,29 @@ func (h *Panos) commit() error {
 		Admins:      []string{h.adminUser},
 		Description: "Consul Terraform Sync Commit",
 	}
-	job, resp, err := h.client.Commit(c.Element(), "", nil)
-	if emptyCommit(job, resp, err) {
+	tryCommit := func() error {
+		job, resp, err := h.client.Commit(c.Element(), "", nil)
+		if emptyCommit(job, resp, err) {
+			return nil
+		}
+		if err != nil {
+			log.Printf("[ERR] (handler.panos) error committing: %s. Server response: '%s'", err, resp)
+			return err
+		}
+
+		if err := h.client.WaitForJob(job, nil); err != nil {
+			log.Printf("[ERR] (handler.panos) error waiting for panos commit to finish: %s", err)
+			return err
+		}
 		return nil
 	}
-	if err != nil {
-		log.Printf("[ERR] (handler.panos) error committing: %s. Server response: '%s'", err, resp)
-		return err
-	}
 
-	if err := h.client.WaitForJob(job, nil); err != nil {
-		log.Printf("[ERR] (handler.panos) error waiting for panos commit to finish: %s", err)
-		return err
+	if err := tryCommit(); err != nil {
```

Comment:
Not sure if you're already aware, but wanted mention that there's an existing [`retry` package](https://github.com/hashicorp/consul-terraform-sync/blob/master/retry/retry.go) that is pretty small and is currently used for [retrying task execution in daemon mode](https://github.com/hashicorp/consul-terraform-sync/blob/master/controller/readwrite.go#L203). It has slightly different logic in that it uses a random delay based on seed. Would you want to repurpose/modifie for this single retry too?

Reply:
I missed retry getting added. I had thought about using a more standard retry algorithm but was waiting to talk to Tristan about it. Now (post meeting) that I have and determined exponential backoff is appropriate I can just use it as is... Thanks!

Also, more importantly IMO, that reminded me we really should be passing the Context through the handler Do() calls to allow for interruptions.

I'm going to rework handlers to pass Context and this case to use that retry package.

Reply:
All done with these changes. Thanks again.